src/JWT/InvalidTokenPartsException.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+
+namespace JWT
+{
+    /// <summary>
+    /// Represents an exception thrown when when a token doesn't consist of 3 delimited by dot parts.
+    /// </summary>
+    public class InvalidTokenPartsException : ArgumentOutOfRangeException
+    {
+        /// <summary>
+        /// Create the SignatureVerificationException.
+        /// </summary>
+        /// <param name="paramName">The name of the parameter that caused the exception.</param>
+        public InvalidTokenPartsException(string paramName)
+            : base(paramName, "Token must consist of 3 delimited by dot parts.")
```

Comment:
Grammar is a little off here. I suggest: "Token must consist of three **parts** delimited by dots."

Reply:
My very first PR to Jwt back few years ago :) Fixed